### PR TITLE
gammu: fix script dependencies on bash and dialog

### PR DESCRIPTION
**Pull request by wamserma**
###### Description of changes

Fixes shebang lines and dependency on a `dialog` program in helper scripts.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/gammu/default.nix [hunk 1]
```diff
@@ -17,7 +17,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-aeaGHVxOMiXRU6RHws+oAnzdO9RY1jw/X/xuGfSt76I=";
   };
 
-  patches = [ ./bashcomp-dir.patch ./systemd.patch ];
+  patches = [ ./bashcomp-dir.patch ./systemd.patch ./gammu-config-dialog.patch ];
+  postPatch = ''
+    substituteInPlace utils/jadmaker --replace "/bin/bash" "${bash}/bin/bash"
+    substituteInPlace utils/gammu-config --replace "/bin/bash" "${bash}/bin/bash"
+    substituteInPlace utils/gammu-config --replace "NIX_PUT_DIALOG_PATH_HERE" "${dialog}/bin/dialog"
```
Comment (Majiir) [inline pkgs/applications/misc/gammu/default.nix:24]: We can use `substituteAll` on the patch itself. Maybe this is cleaner? For example:

```nix
{
  patches = [
    ./bashcomp-dir.patch
    ./systemd.patch
    (substituteAll {
      src = ./gammu-config-dialog.patch;
      dialog-path = "${dialog}/bin/dialog";
    })
  ];
}
```

where the patch would contain a line like `DIALOG=@dialog-path@`.

Comment (Majiir) [reply pkgs/applications/misc/gammu/default.nix:24]: We can also use `patchShebangs` for the other rules. See: https://discourse.nixos.org/t/what-is-the-patchshebangs-command-in-nix-build-expressions/12656#32-explicitly-6

Comment (wamserma) [reply pkgs/applications/misc/gammu/default.nix]: Using substituteAll on the patch is a good idea.

Comment (wamserma) [reply pkgs/applications/misc/gammu/default.nix]: No need to explicitly patch the shebangs. But `bash` has to be included in the `buildInputs` as `strictDeps = true;` is used in the derivation.

File: pkgs/applications/misc/gammu/default.nix [hunk 2]
```diff
@@ -17,15 +17,23 @@ stdenv.mkDerivation rec {
...
 
   nativeBuildInputs = [ pkg-config cmake ];
 
   strictDeps = true;
 
   buildInputs = [ bluez libusb1 curl gettext sqlite libiconv ]
   ++ optionals dbiSupport [ libdbi libdbiDrivers ]
-  ++ optionals postgresSupport [ postgresql ];
+  ++ optionals postgresSupport [ postgresql ]
+  ++ optionals strictDeps [ bash ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/gammu/default.nix:36]: strictDeps conditionals is not allowed. Please include this by default

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/gammu/default.nix (hunk 2 above)
Comment (wamserma) [reply pkgs/applications/misc/gammu/default.nix:36]: Done.

**Pull request merged**